### PR TITLE
util/admin: support admin check index on partition table

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

As title said.

Before this PR, `admin check index` on partition table will panic.

### What is changed and how it works?

This PR removes the `CheckIndexExec ` executor, and use `CheckTableExec` executor to do `admin check index`.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- No

### Release note 

- Support admin check index on the partition table

### CODE REVIEW

**Comment by crazycs520** [comment]
@tiancaiamao @zimulala  PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -1206,6 +1168,79 @@ func (b *PlanBuilder) buildPhysicalIndexLookUpReaders(ctx context.Context, dbNam
...
+		return nil, errors.Errorf("index %s do not exist", as.Index)
+	}
+	if idx.State != model.StatePublic {
+		return nil, errors.Errorf("index %s state %s isn't public", as.Index, idx.State)
+	}
+
+	return b.buildPhysicalIndexLookUpReader(ctx, dbName, tbl, idx)
+}
+
+func (b *PlanBuilder) buildAdminCheckIndex(ctx context.Context, as *ast.AdminStmt) (*CheckTable, error) {
```
Comment (tangenta) [inline planner/core/planbuilder.go:1243]: Where is the usage of this function?

Comment (crazycs520) [reply planner/core/planbuilder.go]: great, done.

**Pull request merged**